junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,522 @@
+package org.scalajs.junit.plugin
+
+import scala.language.reflectiveCalls
+
+import scala.reflect.internal.Flags
+import scala.tools.nsc._
+import scala.tools.nsc.plugins.{
+  Plugin => NscPlugin, PluginComponent => NscPluginComponent
+}
+
+/** The Scala.js jUnit plugin is a way to overcome make class annotation
+ * information (usually accessed by reflection) of any test class accessible
+ * to the Scala.js testing framework.
+ *
+ *
+ * As an example we take the following test class:
+ * {{{
+ * class Foo {
+ *   @Before def before(): Unit = {
+ *     // Initialize the instance before the tests
+ *   }
+ *   @Test def bar(): Unit = {
+ *     // assert some stuff
+ *   }
+ *   @Ignore("baz not implemented yet") @Test def baz(): Unit = {
+ *     // assert some other stuff
+ *   }
+ * }
+ * object Foo {
+ *   @BeforeClass def beforeClass(): Unit = {
+ *     // Initialize some global state for the tests.
+ *   }
+ * }
+ * }}}
+ *
+ * Will be transformed to:
+ *
+ * {{{
+ * class Foo extends org.scalajs.junit.ScalaJSJUnitTest {
+ *   @Before def before(): Unit = {
+ *     // Initialize the instance before the tests
+ *   }
+ *   @Test def bar(): Unit = {
+ *     // assert some stuff
+ *   }
+ *   @Ignore @Test def baz(): Unit = {
+ *     // assert some other stuff
+ *   }
+ *   def scalajs\$junit\$invoke(methodId: String): Unit = {
```

Comment:
Nice! That is an awesome idea. Actually we don't even need to type `instance` as `AnyRef`, but we can actually use the full type of the test. This can then be an abstract type member in `JUnitTestBootstrapper` and everything is super neat! (also, it would be really, really nice, if we could make this `methodId` an `Int` :))